src/main/java/io/cdap/plugin/SCPRemotetoRemoteAction.java

```diff
@@ -0,0 +1,293 @@
+     * is a good place to validate any configuration options the user has entered. If this throws
+     * an exception, the Pipeline will not be deployed and the user will be shown the error message.
+     */
+    @Override
+    public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+        super.configurePipeline(pipelineConfigurer);
+        LOG.debug(String.format("Running the 'configurePipeline' method of the %s plugin.", PLUGIN_NAME));
+        config.validate();
+    }
+
+    @Override
+    public void run(ActionContext context) throws JSchException, IOException {
+        LOG.debug(String.format("Running the 'run' method of the %s plugin.", PLUGIN_NAME));
+        config.validate();
+
+        byte[] key = config.getPrivateKey();
+        byte[] passphrase = config.getPassphrase();
+        JSch jsch = new JSch();
+        jsch.addIdentity("key", key, null,passphrase);
+        Session session = jsch.getSession(config.getUserNameBastion(),config.getHostBastion(),config.getPort());
+
+        session.setConfig("StrictHostKeyChecking", "no");
+        session.connect();
+        ChannelExec channel = (ChannelExec) session.openChannel("exec");
+
+        String userA = config.getUserNameA();
+        String hostA = config.getHostA();
+        String source = config.getSource();
+        String pathSource = userA+"@"+hostA+":"+source;
+
+        String userB = config.getUserNameB();
+        String hostB = config.getHostB();
+        String dest= config.getDest();
+        String pathDest = userB+"@"+hostB+":"+dest;
+
+        String dirFlag = config.getDirFlag();
+        String compressFlag = config.getCompressFlag();
+        String verboseFlag = config.getVerboseFlag();
+
+        //Host A -> Host B
+        channel.setCommand("scp "+compressFlag+ " "+verboseFlag+
+            " " +dirFlag +" " +pathSource +" " +pathDest);
+
+        StringBuilder outputBuffer = new StringBuilder();
+        StringBuilder errorBuffer = new StringBuilder();
+
+        InputStream in = channel.getInputStream();
+        InputStream err = channel.getExtInputStream();
+
+        channel.connect();
+
+        byte[] tmp = new byte[1024];
+        while (true) {
+            while (in.available() > 0) {
+                int i = in.read(tmp, 0, 1024);
+                if (i < 0) break;
+                outputBuffer.append(new String(tmp, 0, i));
+            }
+            while (err.available() > 0) {
+                int i = err.read(tmp, 0, 1024);
+                if (i < 0) break;
+                errorBuffer.append(new String(tmp, 0, i));
+            }
+            if (channel.isClosed()) {
+                if ((in.available() > 0) || (err.available() > 0)) continue;
+                System.out.println("exit-status: " + channel.getExitStatus());
+                break;
+            }
+            try {
+                Thread.sleep(1000);
+            } catch (Exception ignored) {
```

Comment:
Shouldn't ignore interrupted exception. Upon interruption, it should quite the loop